core/controllers/reader.py

```diff
@@ -79,6 +79,40 @@ def _does_exploration_exist(exploration_id, version, collection_id):
 class ExplorationEmbedPage(base.BaseHandler):
     """Page describing a single embedded exploration."""
 
+    URL_PATH_ARGS_SCHEMAS = {
+        'exploration_id': {
+            'schema': {
+                'type': 'basestring'
+            }
+        }
```

Comment:
Please validate the exploration ID format. You can look into this or other files to see how it is done.

Reply:
Done